### ISSUE TITLE
Empty required fields breaking my contact form

### ISSUE THREAD

**Issue by mortscode** [craftcms/contact-form#115]
Attempting to add custom inputs for Phone Number and a hidden checkbox to describe which form of multiple forms from my site that the current user is using. I'm breaking apart the message field into an array as described in the documentation (`name=message[Phone]`). If all of the required fields are filled, everything works swimmingly. However, if any of the required fields are empty, I get an `Array to String Conversion` error. On a form that uses no custom inputs, the errors return as expected.

All errors work as expected until I change the `name` attribute of the message field to `name=message[body]` as described in the documentation.

### To Reproduce
1. Add custom labels using the message array method from the docs (i.e. `name="message[phone]"` field)
2. Change the message `<textarea>` name tag to `name="message[body]"`
3. Navigate to the form and click "Submit" with nothing in the fields

- Expecting error messages under required fields
- Receiving `Array to String Conversion` error instead

### Template code from one of the breaking forms
```
{% macro errorList(errors) %}
    {% if errors %}
        <ul class="errors">
            {% for error in errors %}
                <li>{{ error }}</li>
            {% endfor %}
        </ul>
    {% endif %}
{% endmacro %}

{% from _self import errorList %}

<form method="post" action="" accept-charset="UTF-8" class="start__form--form">
    {{ csrfInput() }}
    <input type="hidden" name="action" value="contact-form/send">
    {{ redirectInput('contact/thanks') }}

    <label class="start__form--user-type"><input  type="checkbox" name="message[Type][]" value="New to CrossFit" checked> New to CrossFit</label>

    <div class="start__form--form-field">
      <h3 class="start__form--label"><label for="from-name">Your Name</label></h3>
      <input id="from-name" class="start__form--input" type="text" name="fromName" value="{{ message.fromName ?? '' }}">
    </div>
    {{ message is defined and message ? errorList(message.getErrors('fromName')) }}

    <div class="start__form--form-field">
      <h3 class="start__form--label"><label for="from-email">Email Address</label></h3>
      <input id="from-email" class="start__form--input" type="email" name="fromEmail" value="{{ message.fromEmail ?? '' }}">
    </div>
    {{ message is defined and message ? errorList(message.getErrors('fromEmail')) }}

    <div class="start__form--form-field">
      <h3 class="start__form--label"><label for="phone">Phone Number</label></h3>
      <input id="phone" class="start__form--input" type="text" name="message[phone]" value="">
    </div>
    {{ message is defined and message ? errorList(message.getErrors('message')) }}

    <div class="start__form--form-field">
        <h3 class="start__form--label"><label for="message">Message</label></h3>
        <textarea id="message" class="start__form--textarea" rows="10" name="message[body]" placeholder="What’s your primary motivation for starting CrossFit?">{{ message.message ?? '' }}</textarea>
    </div>
    {{ message is defined and message ? errorList(message.getErrors('message')) }}

    <input id="pullups" name="pullups" type="text">
    
    <div class="start__form--submit-wrapper">
      <input class="start__form--submit button -green-dark" type="submit" value="Send">
    </div>
</form>
```

**Comment by angrybrad** [comment]
```
        <textarea id="message" class="start__form--textarea" rows="10" name="message[body]" placeholder="What’s your primary motivation for starting CrossFit?">{{ message.message ?? '' }}</textarea>
```

Should be:

```
        <textarea id="message" class="start__form--textarea" rows="10" name="message[body]" placeholder="What’s your primary motivation for starting CrossFit?">{{ message.message['body'] ?? '' }}</textarea>
```

**Comment by mortscode** [comment]
Works like a dream.

Follow up question, is there a way to pass contact form vars into the
“Subject Text” on the Contact Form settings? Right now, all of the contact
form submissions come in with the same Subject, which ends up making them
one long thread in my inbox.

Any thoughts?


—

*Mort*
Developer
971.258.1113

Instrument
instrument.com

On Feb 18, 2018 at 5:37 PM, Brad Bell <notifications@github.com> wrote:


Closed #115 <https://github.com/craftcms/contact-form/issues/115>.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/craftcms/contact-form/issues/115#event-1479760891>, or mute
the thread

**Comment by mortscode** [comment]
I should specify that the Subject field has been removed from 2 of the 3
contact forms on the site, I’d love to pass a hidden subject with the
contact form type (new athlete, experienced athlete, etc) followed by the
entered Full Name.

Is it possible to do that?


—

*Mort*
Developer
971.258.1113

Instrument
instrument.com

On Feb 19, 2018 at 9:46 PM, Mort <mort@instrument.com> wrote:


Works like a dream.

Follow up question, is there a way to pass contact form vars into the
“Subject Text” on the Contact Form settings? Right now, all of the contact
form submissions come in with the same Subject, which ends up making them
one long thread in my inbox.

Any thoughts?


—

*Mort*
Developer
971.258.1113

Instrument
instrument.com


On Feb 18, 2018 at 5:37 PM, Brad Bell <notifications@github.com> wrote:


Closed #115 <https://github.com/craftcms/contact-form/issues/115>.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/craftcms/contact-form/issues/115#event-1479760891>, or mute
the thread
